mono/mini/interp/interp.c

```diff
@@ -4032,28 +4011,16 @@ call:;
 			 */
 			SAVE_INTERP_STATE (frame);
 
-			child_frame = alloc_frame (context, native_stack_addr, frame, cmethod, sp, retval);
-
-			if (G_UNLIKELY (!cmethod->transformed)) {
-				MonoException *ex;
-				gboolean tracing;
+			frame = alloc_frame (context, native_stack_addr, frame, cmethod, sp, retval);
 
-				method_entry (context, child_frame, &tracing, &ex);
+			gboolean tracing;
 
-				if (G_UNLIKELY (ex)) {
-					frame = child_frame;
-					frame->ip = NULL;
+			if (method_entry (context, frame, &tracing, &ex)) {
+				if (ex)
 					THROW_EX (ex, NULL);
-					EXCEPTION_CHECKPOINT;
-				}
-			} else {
-				alloc_stack_data (context, child_frame, cmethod->alloca_size);
-#if DEBUG_INTERP
-				debug_enter (child_frame, &tracing);
-#endif
+				EXCEPTION_CHECKPOINT;
```

Comment:
Moving `EXCEPTION_CHECKOUT` out of the `if (ex)` is important, at least sometimes.

Reply:
This is how we used to do it : https://github.com/mono/mono/blob/2019-10/mono/mini/interp/interp.c#L3243

It probably regressed with the nonrecursive change.

Reply:
Right, the recursive case is still ok, not changed here, that is why newobj_fast was passing the test before this PR.

Reply:
And right, I kept looking at the recursive and non-recursive cases until I finally noticed the difference. Debugging the unwinding wasn't seeming fruitful.